src/dotnet/APIView/APIViewWeb/Managers/ReviewManager.cs

```diff
@@ -483,9 +660,137 @@ private async Task<ReviewListItemModel> ToggleReviewApproval(ClaimsPrincipal use
+                    .Where(r => r.NamespaceApprovalRequestedOn.HasValue &&
+                               Math.Abs((r.NamespaceApprovalRequestedOn.Value - namespaceTimestamp).TotalSeconds) < 5 &&
+                               !r.IsDeleted)
+                    .ToList();
+
+                return matchingReviews;
+            }
+            catch (Exception ex)
+            {
+                _telemetryClient.TrackException(ex);
+                return new List<ReviewListItemModel>();
+            }
+        }
+
+        /// <summary>
+        /// Find related reviews (SDK languages + TypeSpec) via pull request relationships
+        /// </summary>
+        /// <param name="reviewId">The review ID to find related reviews for</param>
+        /// <returns>List of related reviews (SDK + TypeSpec) via pull requests, or empty list if none found</returns>
+        private async Task<List<ReviewListItemModel>> FindRelatedReviewsByPullRequest(string reviewId)
+        {
+            var relatedReviews = new List<ReviewListItemModel>();
+            
+            try
+            {
+                var relatedReviewIds = await FindRelatedReviewsByPullRequestAsync(reviewId);
+                
+                foreach (var relatedId in relatedReviewIds)
+                {
+                    try
+                    {
+                        var review = await _reviewsRepository.GetReviewAsync(relatedId);
+                        if (review != null && 
+                            LanguageHelper.IsSDKLanguageOrTypeSpec(review.Language) &&
+                            !review.IsDeleted)
+                        {
+                            relatedReviews.Add(review);
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        _telemetryClient.TrackException(ex);
+                    }
+                }
+                
+                return relatedReviews;
+            }
+            catch (Exception ex)
+            {
+                _telemetryClient.TrackException(ex);
+                return new List<ReviewListItemModel>();
+            }
+        }
+
+        /// <summary>
+        /// Check if all related SDK reviews are approved and if so, approve namespace for TypeSpec review
+        /// Uses namespace request timestamp to efficiently check if all related reviews are approved
+        /// </summary>
+        /// <param name="approvedReview">The review that was just approved</param>
+        /// <returns></returns>
+        private async Task CheckAndApproveNamespaceForTypeSpec(ReviewListItemModel approvedReview)
+        {
+            try
+            {
+                var namespaceRequestTimestamp = approvedReview.NamespaceApprovalRequestedOn.Value;
+
+                // Check if any reviews with the same timestamp are still pending
+                var pendingSdkReviews = await GetReviewsWithNamespaceTimestamp(namespaceRequestTimestamp);
```

Comment:
is the timestamp the only or best way to do the relation on which is the corresponding review?